docs/build.md

```diff
@@ -1,8 +1,126 @@
-* Build custom software
-
-The build code keeps a cache of all packages source code (Git/Mercurial/SVN)
-repositories so that they do not need to be downloaded on every build. If it is
-desired to change the source code of a package, you may edit the package local
-repository content. To build the package from this modified local repository,
-either use a file:// URL pointing to it in the package metadata (YAML)
-file or disable package repositories update in config/host_os.yaml.
+# Building process
+
+The build process relies on a metadata
+[repository](https://github.com/open-power-host-os/versions) to obtain
+information about the components to build, such as URLs, versions,
+commit IDs, dependencies and distro-specific build parameters.
+
+During the build process, the build script checks out the metadata
+(versions) repository, obtains the information necessary to build the
+package and starts an instance of
+[mock](https://github.com/rpm-software-management/mock/wiki) that
+builds the package in a chroot, taking care of its build dependencies.
+
+Other functionalities of the build script are:
+
+  - [build an installation ISO image](building_an_iso.md)
+  - update the metadata in the versions repository according to each
+   of the package's source code upstream
+  - update the open-power-host-os metapackages
+  - publish a summary of the code updates to the [web page](https://open-power-host-os.github.io)
+
+
+Some common use cases when building packages are:
+
+  1. Building one or a number of packages without modification from
+  what the `versions` repository provides;
+
+  1. Building a specific version of a package different from the one
+    listed in the `versions` repository.
+
+  1. Building a modified version of a package.
+
+----
+*For the following sections, make sure to have executed the initial setup:*
+
+```
+git clone https://github.com/open-power-host-os/builds.git
+cd builds
+
+sudo yum install epel-release
+sudo yum install -y $(cat rpm_requirements.txt)
+sudo usermod -a -G mock $(whoami)
+```
+----
+
+## Building packages
+
+Building packages is as simple as:
+
+```
+./host_os.py build-packages
+```
+
+which builds all of the packages listed in the
+[versions](open-power-host-os/versions) repository or:
+
+```
+./host_os.py build-packages --packages kernel
+```
+
+which builds a specific package or list of packages.
+
+The built packages are kept in a `result` directory (which can be
+changed with the `--result-dir` option), and are not rebuilt in
+subsequent runs of the command unless their source file have changed
+since the creation of the last build results (to avoid this, use
+--force-rebuild).
```

Comment:
This is basically a copy of the quick start guide. Should we really keep this info duplicated? It will likely lead to inconsistent docs in the future.